### ISSUE TITLE
Adding graphql_client as dependency in Router Plugin causes internal dependency issue

### ISSUE THREAD

**Issue by BCantos17** [apollographql/router#937]
**Describe the bug**
I am writing a plugin to use the graphql client to make a query call for authorization but when I add graphql_client as a dependency, it breaks the the code in https://github.com/apollographql/rover/blob/geal/launchpad/crates/launchpad/src/introspect/runner.rs

**To Reproduce**
1. set up context example project -> https://github.com/apollographql/router/tree/v0.1.0-preview.6/examples/context
2. add these as dependency in Cargo.toml 
````
graphql_client = { version = "0.10.0", features = ["reqwest-blocking"]}
reqwest = { version = "0.11.10", features = ["blocking"]}
````
3. run `cargo build`

**Expected behavior**
cargo build compiles without any issues

**Output**
````
  Compiling launchpad v0.1.0 (https://github.com/apollographql/rover.git?rev=94141242ba34cf00cde9630fc4a6dcd05d4fa5da#94141242)
error[E0433]: failed to resolve: use of undeclared type `HeaderMap`
  --> /Users/macuser/.cargo/git/checkouts/rover-efd9f422be37a06b/9414124/crates/launchpad/src/introspect/runner.rs:30:26
   |
30 |     let mut header_map = HeaderMap::new();
   |                          ^^^^^^^^^ not found in this scope
   |
help: consider importing this struct
   |
1  | use reqwest::header::HeaderMap;
   |

error[E0433]: failed to resolve: use of undeclared type `HeaderName`
  --> /Users/macuser/.cargo/git/checkouts/rover-efd9f422be37a06b/9414124/crates/launchpad/src/introspect/runner.rs:33:13
   |
33 |             HeaderName::from_bytes(header_key.as_bytes())?,
   |             ^^^^^^^^^^ not found in this scope
   |
help: consider importing this struct
   |
1  | use reqwest::header::HeaderName;
   |

error[E0433]: failed to resolve: use of undeclared type `HeaderValue`
  --> /Users/macuser/.cargo/git/checkouts/rover-efd9f422be37a06b/9414124/crates/launchpad/src/introspect/runner.rs:34:13
   |
34 |             HeaderValue::from_str(&header_value)?,
   |             ^^^^^^^^^^^ not found in this scope
   |
help: consider importing this struct
   |
1  | use reqwest::header::HeaderValue;
   |

error[E0659]: `reqwest` is ambiguous
 --> /Users/macuser/.cargo/git/checkouts/rover-efd9f422be37a06b/9414124/crates/launchpad/src/introspect/runner.rs:6:5
  |
6 | use reqwest::header::{HeaderMap, HeaderName, HeaderValue};
  |     ^^^^^^^ ambiguous name
  |
  = note: ambiguous because of multiple potential import sources
  = note: `reqwest` could refer to a crate passed with `--extern`
  = help: use `::reqwest` to refer to this crate unambiguously
note: `reqwest` could also refer to the module imported here
 --> /Users/macuser/.cargo/git/checkouts/rover-efd9f422be37a06b/9414124/crates/launchpad/src/introspect/runner.rs:5:5
  |
5 | use graphql_client::*;
  |     ^^^^^^^^^^^^^^^^^
  = help: use `self::reqwest` to refer to this module unambiguously

Some errors have detailed explanations: E0433, E0659.
For more information about an error, try `rustc --explain E0433`.
error: could not compile `launchpad` due to 4 previous errors
````

**Additional context**
Dependency tree is apollo router -> uplink -> rover (branch = geal/launchpad)

**Comment by Geal** [comment]
I see, this should be fixed by https://github.com/apollographql/rover/pull/1061/commits/1123e3f14d0750cda1b645416fd84ea740df57a6 and will be available shortly in the router

**Comment by BCantos17** [comment]
thanks @Geal! Any idea if this will be available in another preview? If not then I will test off of main

**Comment by Geal** [comment]
We're doing a new release this week, it should be available soon

**Comment by abernix** [comment]
@BCantos17 There's a  new release out which should have this fix in it! [Found here as `v0.1.0-preview.7`](https://github.com/apollographql/router/releases/tag/v0.1.0-preview.7)
